hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -813,7 +813,9 @@ public Response putBucketLifecycleConfiguration(
       if (ex.getResult() == ResultCodes.ACCESS_DENIED) {
         throw S3ErrorTable.newError(S3ErrorTable.ACCESS_DENIED, bucketName);
       } else if (ex.getResult() == ResultCodes.INVALID_REQUEST) {
-        throw S3ErrorTable.newError(S3ErrorTable.INVALID_REQUEST, bucketName);
+        OS3Exception invalidException = S3ErrorTable.INVALID_REQUEST;
+        invalidException.setErrorMessage(ex.getMessage());
+        throw S3ErrorTable.newError(invalidException, bucketName);
```

Comment:
Nit: Add `setResource(bucketName)`

Reply:
This `OS3Exception#resource` is not used in `newError`, so adding `setResource` here is invalid.